session/session.go

```diff
@@ -799,13 +800,18 @@ func (s *session) ExecRestrictedSQLWithSnapshot(sql string) ([]chunk.Row, []*ast
 	}
 	// Set snapshot.
 	if snapshot != 0 {
+		se.sessionVars.SnapshotInfoschema, err = domain.GetDomain(s).GetSnapshotInfoSchema(snapshot)
```

Comment:
Is it possible that "`se.sessionVars.SnapshotInfoschema`" is not `nil` here?

Reply:
Since we reset `SnapshotInfoschema` after putting se back to the pool, it should always be nil.